Firestore/core/src/api/query_core.cc

```diff
@@ -238,22 +239,28 @@ Query Query::Filter(FieldPath field_path,
           "ID since document IDs are not arrays.",
           Describe(op));
     } else if (op == Operator::In || op == Operator::NotIn) {
-      ValidateDisjunctiveFilterElements(value, op);
-      google_firestore_v1_ArrayValue& array = value.array_value;
-      for (pb_size_t i = 0; i < array.values_count; ++i) {
-        array.values[i] =
-            ParseExpectedReferenceValue(array.values[i], type_describer);
-      }
+      ValidateDisjunctiveFilterElements(*value, op);
+      google_firestore_v1_Value references{};
```

Comment:
Question: is creating a separate value and copying elements into it necessary? Can this be done in place?

Reply:
I think nothing would break if we did not create a new array here, but we would be modifying the input array and would have to manually free values from the input array if we change the value type in ParseExpectedReferenceValue. 

I added a TODO since ParseExpectedReferenceValue will return a Message object in a follow up PR. I have to see if I can address your comment easily. In general, this is not code that gets invoked very often.